### PR TITLE
Link depends on whole packages

### PR DESCRIPTION
**Pull request by Julow**
This is on top of https://github.com/jonludlam/odocmkgen/pull/6 but splitted for easier reviewing.

Link dependencies are no longer the compile dependencies but the whole packages instead.
This fixes the broken test added in #6 (reference from page to module)

### CODE REVIEW

File: lib/compile.ml [hunk 1]
```diff
@@ -26,9 +27,16 @@ let compile_fragment all_infos info =
 
   [ Format.asprintf "%a : %a %s" Fpath.pp odoc_path Fpath.pp (Inputs.input_file info) deps_str;
     Format.asprintf "\t@odoc compile --package %s $< %s -o %a" info.package include_str Fpath.pp odoc_path;
-    Format.asprintf "compile : %a" Fpath.pp odoc_path;
+    Format.asprintf "compile-%s : %a" info.package Fpath.pp odoc_path;
```
Comment (jonludlam) [inline lib/compile.ml:30]: We should mark this target as phony, or make it a real (but hidden, usually) stamp file.

File: lib/compile.ml [hunk 2]
```diff
@@ -26,9 +27,16 @@ let compile_fragment all_infos info =
...
   let print_lines = List.iter (Printf.fprintf oc "%s\n") in
-  List.iter (fun inp -> print_lines (compile_fragment inputs inp)) inputs
+  let packages = Inputs.split_packages inputs in
+  List.iter (fun inp -> print_lines (compile_fragment inputs inp)) inputs;
+  let package_rules_s =
+    List.map (fun (pkg, _) -> "compile-" ^ pkg) (StringMap.bindings packages)
+    |> String.concat " "
+  in
+  Printf.fprintf oc "compile : %s\n" package_rules_s;
+  Printf.fprintf oc ".PHONY : %s\n" package_rules_s
```
Comment (jonludlam) [inline lib/compile.ml:42]: Oh, I see you have :-)

File: lib/compile.ml (hunk 1 above)
Comment (Julow) [reply lib/compile.ml:30]: That's done in the function below, where we have the list of those targets.

File: lib/compile.ml (hunk 2 above)
Comment (Julow) [reply lib/compile.ml:42]: :)

**Pull request merged**